pkg/util/fieldindex/register.go

```diff
@@ -152,3 +160,16 @@ func indexImagePullJobActive(c cache.Cache) error {
 		return []string{isActive}
 	})
 }
+
+func indexSideCatSet(c cache.Cache) error {
+	return c.IndexField(context.TODO(), &appsv1alpha1.SidecarSet{}, IndexNameForSideCarSetNamespace, func(rawObj client.Object) []string {
+		obj := rawObj.(*appsv1alpha1.SidecarSet)
+		if obj == nil {
+			return nil
+		}
+		if obj.Spec.Namespace != "" {
+			return []string{obj.Spec.Namespace}
+		}
+		return []string{IndexValueSideCarSetClusterScope}
```

Comment:
since k8s 1.22, k8s will automatically assign [name label](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#automatic-labelling) in the namespace, so if the sidecarset namespace selector select the built-in label, it should also return the namespace index